### PR TITLE
WIP: Develop

### PR DESCRIPTION
**Pull request by bastienterrier**

### CODE REVIEW

File: docs/.vuepress/components/RegisterToken.vue [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  padding-bottom: 5px;
+}
+.success {
+  color: green;
+}
+.warning {
+  color: orange;
+}
+.error--text {
+  color: orange !important;
```
Comment (EmmanuelDemey) [inline docs/.vuepress/components/RegisterToken.vue:66]: Is it necessary the `important` ?

File: docs/.vuepress/components/RegisterToken.vue [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      ],
+      tokenRules: [v => !!v || 'field required'],
+    };
+  },
+  methods: {
+    registerToken() {
+      if (!this.$refs.form.validate()) {
+        return;
+      }
+      const serverURL = 'http://localhost:3000/config-env';
```
Comment (EmmanuelDemey) [inline docs/.vuepress/components/RegisterToken.vue]: Should we hardcode this URL ? what about the same application deployed elsewhere ?

File: docs/README.md [hunk 3]
```diff
@@ -7,3 +7,15 @@
 This [NestJs](https://docs.nestjs.com/) API expose a set of customizable Rules to automate your project's life cycle.
 
 Keywords : [#NestJs](), [#NodeJs](), [#TypeScript](), [#Docker](), [#Webhooks](), [#FaaS](), [#Github](), [#GitLab](), [#CI](), [#Open-Source]()...
+
+## Prerequisite
+
+This API is available for [Github](https://github.com) and [Gitlab](https://about.gitlab.com/) repositories.
+
+Others Git version control system will may be available in the future. Do not hesitate to ask for one of them.
```
Comment (EmmanuelDemey) [inline docs/README.md]:
```suggestion
Other Git version control systems will may be available in the future. Do not hesitate to ask for one of them.
```

File: docs/guide/gettingStarted.md [hunk 4]
```diff
@@ -1,16 +1,77 @@
 # Getting Started
 
+## Directory structure
+
+If you want to use the `Git Webhooks` API, you need to create a `.git-webhooks` folder in your root directory.
+
+This folder will be source everytime you interact with our API.
```
Comment (EmmanuelDemey) [inline docs/guide/gettingStarted.md]: `source` ?

File: docs/post-actions/customisablePostActions.md [hunk 5]
```diff
@@ -6,38 +6,34 @@ This allow you to customise your process, according to the result of a rule.
...
 - and much more!
 
 ## How does it work?
 
 In your `rules.yml` config file, you can add `callback` functions that will be called if the rule succeeds or not.
 
-These `callback`s are `Runnable` classes, that implementing the `RunnableInterface` interface.
-The `Runnable` interface will have a `run()` method with `RuleResult` and `args` as arguments.
+These `callback`s are `Runnable` classes, that implementing the `Runnable` abstract class.
+The `Runnable` interface will have a `run()` method with `callbackType`, `RuleResult` and `args` as arguments.
```
Comment (EmmanuelDemey) [inline docs/post-actions/customisablePostActions.md]: `Runnable` is an interface or an abstract class ?

File: docs/post-actions/customisablePostActions.md [hunk 6]
```diff
@@ -6,38 +6,34 @@ This allow you to customise your process, according to the result of a rule.
...
 
+`callbackType` precise if the Runnable is called because the rule failed, succeed or if it always called.
 [`RuleResult`](../rules/customisableRules.html#validate-method) is an object containing the result of the rule, and `args` is a custom object, with as many properties as you want.
 
 ## Create your own Runnable class
 
-If you don't find a suitable Post-Action for your needs, you can easily create yours by implementing the `RunnableInterface`.
+If you don't find a suitable Post-Action for your needs, you can easily create yours by extende the `Runnable` class.
```
Comment (EmmanuelDemey) [inline docs/post-actions/customisablePostActions.md]: class or interface ?

File: docs/post-actions/existingRunnables.md [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+
+The `LoggerRunnable` Post-Action, like its name suggest, log informations, using [winston logger](https://github.com/winstonjs/winston).
+
+### Usage
+
+This Post-Action need two args:
+
+- `type`: info|warn _[optional]_,
+  ::: tip
+  If you're in an `onSuccess` or `onBoth` statement, the default value will be `info`. On the other hand, if you're in an `onError` statement, it would be `warn`.
```
Comment (EmmanuelDemey) [inline docs/post-actions/existingRunnables.md]: Why `warn` as a default level for errors ? Why not `error` ?

File: docs/rules/customisableRules.md [hunk 8]
```diff
@@ -1,66 +1,44 @@
 # Customisable Rules
 
-You can customise the existing rules, by adding yours. Rules are quiet simply to extend.
-They are in `rules.yml` file, located in `src/rules/rules.yml` and must respect the `Rule` class (`src/rules/rule.class.ts`).
+The `.git-webhooks/rules.yml` file contains all the rules you've configured.
+Check our example `rules.yml` file [here](./rulesExample.md).
 
-## Rules config file
+You can use our existing rules or adding yours. Rules are very simply to extend.
```
Comment (EmmanuelDemey) [inline docs/rules/customisableRules.md]: `add` ?

File: docs/rules/customisableRules.md [hunk 9]
```diff
@@ -72,35 +50,35 @@ Post-actions `args` support templating: **_Git Webhooks_** use [mustache js](htt
...
 ```
 
 For more examples, check out the [documentation](https://github.com/janl/mustache.js#templates) provide by mustache.
 
----
-
 ## Create your own rule
 
-If you want to create a new type of rule, you can create your own class. It must extend the abstract `Rule` class, and implement the `valide()` method. This method contains all your business logic.
+If you want to create a new type of rule, you can create your own class. This class must extend the abstract `Rule` class, and implement the `valide()` method. This method contains all your business logic.
```
Comment (EmmanuelDemey) [inline docs/rules/customisableRules.md]:
```suggestion
If you want to create a new type of rule, you can create your own class. This class must extend the abstract `Rule` class, and implement the `validate()` method. This method contains all your business logic.
```

File: package.json [hunk 10]
```diff
@@ -36,7 +36,9 @@
     "@nestjs/testing": "6.0.2",
     "@nestjs/websockets": "6.0.2",
     "class-transformer": "0.2.0",
+    "cors": "^2.8.5",
```
Comment (EmmanuelDemey) [inline package.json]: Delete the `^`

File: src/app.controller.spec.ts [hunk 11]
```diff
@@ -1,132 +1,7 @@
...
-      expect(gitlabService.updateCommitStatus).not.toBeCalled();
-    });
-  });
-
   describe('webhook', () => {
-    it('should not call the updateCommitStatus Github nor updateCommitStatus Gitlab services', () => {
-      appController.processWebhook(githubBranchWebhook, res);
-      expect(githubService.updateCommitStatus).not.toBeCalled();
-      expect(gitlabService.updateCommitStatus).not.toBeCalled();
+    it('should not do anything', () => {
```
Comment (EmmanuelDemey) [inline src/app.controller.spec.ts:3]: 🙃?

File: docs/.vuepress/components/RegisterToken.vue (hunk 1 above)
Comment (bastienterrier) [reply docs/.vuepress/components/RegisterToken.vue:66]: yes

File: docs/.vuepress/components/RegisterToken.vue (hunk 2 above)
Comment (bastienterrier) [reply docs/.vuepress/components/RegisterToken.vue]: The Git-Webhooks url is now located in the Vuepress config.js file

File: docs/post-actions/customisablePostActions.md (hunk 5 above)
Comment (bastienterrier) [reply docs/post-actions/customisablePostActions.md]: Class indeed

**Review by EmmanuelDemey — APPROVED** [review]

**Pull request merged**